src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2952,7 +2941,7 @@ private bool IsEndOfNameInExplicitInterface()
                     // Use else if, rather than if, because if we see both a constructor initializer and a constraint clause, we're too lost to recover.
                     var colonToken = this.CurrentToken;
                     // Set isStatic to false because pretending we're in a static constructor will just result in more errors.
-                    ConstructorInitializerSyntax initializer = this.ParseConstructorInitializer(identifier.ValueText, isStatic: false);
+                    var initializer = this.ParseConstructorInitializer();
```

Comment:
I can't actually tell what the compiler rule on typing is.  "ParseConstructorDeclaration", for example, uses 'var' for everything but the constructor initializer.  It seems like there's little (if any) consistency here.  I can put the types back in, but it seems like it makes things worse instead of bringing a consistent style to these methods.